target/linux/realtek/files-6.6/drivers/net/dsa/rtl83xx/dsa.c

```diff
@@ -1456,10 +1457,16 @@ static int rtl83xx_vlan_add(struct dsa_switch *ds, int port,
 
 	mutex_lock(&priv->reg_mutex);
 
-	if (vlan->flags & BRIDGE_VLAN_INFO_PVID)
+	if (dsa_port_is_cpu(dp)) {
+		if ((priv->ports[port].pvid == RTL_DEFAULT_VID) || (vlan->vid < priv->ports[port].pvid)) {
```

Comment:
Sorry, but this is crazy.  Completely arbitrary and unexpected outcome.

Reply:
patch removed from last PR. wont fix